src/vtk_calculator.py

```diff
@@ -44,25 +44,48 @@ def parse_args():
                         help="The mesh (VTK Unstructured Grid) used as input")
     parser.add_argument("--function", "-f", dest="function", required=True,
                         help="""The function to evalutate on the mesh.
-            Syntax is the same as used in the calculator object, coordinates are given as e.g.  'cos(x)+y'.""")
+            Syntax is the same as used in the calculator object, coordinates are given as e.g.  'cos(x)+y'.
+            Alternatively, you can use predefined function visit preCICE documentation.""")
     parser.add_argument("--output", "-o", dest="out_meshname", default=None, help="""The output meshname.
             Default is the same as for the input mesh""")
     parser.add_argument("--data", "-d", dest="data", default="MyData", help="""The name of output data.
-            Default is MyData""")
+            Default is \"MyData\".""")
     parser.add_argument("--diffdata", "-diffd", dest="diffdata", help="""The name of difference data.
             Used in diff mode. If not given, output data is used.""")
     parser.add_argument("--log", "-l", dest="logging", default="INFO",
                         choices=["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"], help="""Set the log level.
             Default is INFO""")
     parser.add_argument("--directory", "-dir", dest="directory", default=None,
                         help="Directory for output files (optional)")
-    parser.add_argument("--diff", action='store_true', help="Calculate the difference to present data.")
+    parser.add_argument("--diff", action='store_true',
+                        help="Calculate the difference to present data.")
     parser.add_argument("--stats", "-s", action='store_true',
                         help="Store stats of the difference calculation as the separate file inputmesh.stats.json")
     args = parser.parse_args()
     return args
 
 
+preDefFunctions = {
+    "franke3d": """0.75*exp(((9*x-2)^2+(9*y-2)^2+(9*z-2)^2)/4)+0.75*exp(-(9*x+1)^2/49-(9*y+1)/10-(9*z+1)/10)
```

Comment:
```suggestion
    "franke3d": """0.75*exp(-((9*x-2)^2+(9*y-2)^2+(9*z-2)^2)/4)+0.75*exp(-(9*x+1)^2/49-(9*y+1)/10-(9*z+1)/10)
```
Please check again. I think there is a minus missing.

Reply:
Now all correct double-checked. 2D functions are generated from another dictionary, reduces duplication.